### ISSUE TITLE
Crashes always crash the application

### ISSUE THREAD

**Issue by hekhy** [flutter/flutter#90585]
As the title says now since the update to flutter 2.5 whenever a mistake is made in the code that is not indicated in the editor with a red mark, the application crashes completely instead of showing you the exception in console as it has always done . That is very annoying because you always have to hit the play button until the application gets good and thus be able to do a hot restart or you have to stop it completely and start it again.

For instance, every time I change a StatelessWidget for a StatefulWidget and save my application it crashes because it tries to do the hot reload, and I have to hit the play button until it finally lets me restart the app. If I didn't restart it like this, I would have to stop it and restart completely because it won't let me continue. This is something very annoying that did not happen before, so it would be good if you did something about it, because obviously I already know that if I have changed a StatelessWidget for a StatefulWidget that can cause me problems and I have to restart but anyway it is not necessary to stop all my application but to indicate it to me by console as it has always done and so I fix it before continuing again, since now when it stops it makes you waste a lot of time ...

**Comment by darshankawar** [comment]
@hekhy 
Please provide `flutter doctor -v` and a [minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that we can use, to verify the issue you are facing.

You provided a description of the behavior you are seeing, but in order to make this issue actionable, we need steps to replicate along with reproducible code sample to work with.

Thanks.

**Comment by hekhy** [comment]
I'm not going to do that, I've already explained the problem to you well. I have also put a very clear working example, you only have to generate the app and start it in the emulator, once it is started you change a StatelessWidget in your code for a StatefulWidget and save the file, the application automatically stops.

**Comment by darshankawar** [comment]
@hekhy 
Thanks for the update.  Based on details you provided above, I tried to replicate it, but was unable to see any issues with it.
After updating `StatelessWidget` to `Stateful` and saving the changes, the app remain intact, didn't close or quit / stop. There weren't any logs in the console too:

![run](https://user-images.githubusercontent.com/67046386/134663991-e55eed12-b444-4a43-a5e0-f10265a20424.png)

I did this test on latest stable 2.5.1.

If you have specific steps to replicate the issue, please provide us the same.

**Comment by hekhy** [comment]
https://user-images.githubusercontent.com/70041483/134675688-cf01b38a-88b9-4153-8011-2c8517c3f05a.mp4

Here is a video that shows what I am saying. I'm also on flutter 2.5.1, I'm surprised that this doesn't happen to you, or if it happens to you, but as in your screenshot you only show the bottom part and you don't show the debugging controls, which is what I do. I am referring and have already said it several times. But well in the video you can see how the debugging stops therefore my application crashes until I restart it. So I have to hit the play button in the debugging controls or emulator controls, whatever you like to call them more, but if I don't hit play all those times when I hit hot restart it stays in the attempt showing all the time the message "Performing hot restart" at the bottom right, which would force me to completely stop debugging and start again to continue working.
As I have already said this I did not flutter before, I do not understand why we have gone backwards in that.

**Comment by darshankawar** [comment]
@hekhy 
Thanks for the details. As I see from the video, it throws `myApp` is not a subtype of StatelessWidget in type cast` exception.
When we change from `Stateless` to `Stateful`,  we need to do a hot restart, since this change is invoked in main. If we `hot reload`, it directly calls the `build()` function and hence the exception. We need to be able to call `initState()` which gets called after hot restarting the app after making the state change.

Can you try the same, ie, after changing from stateless to stateful, save and then hot restart and see if you get same error or not ?

**Comment by hekhy** [comment]
> Can you try the same, ie, after changing from stateless to stateful, save and then hot restart and see if you get same error or not ?

Yes, of course, but I can tell you that it just keeps trying to do the "hot restart" for an infinite time.
Something that I mentioned in the previous answer, but I will also make a video so you can see it.

https://user-images.githubusercontent.com/70041483/135094857-400b9490-2670-49d9-8117-1dc470a47f4f.mp4

As you can see, he keeps trying to do a hot restart but he never succeeds, in the video I only left him a little time for not making it long but I tried it before leaving it even 10 minutes and it continued like this.
Then as you can see I have to completely stop the application to continue with my work, but if it is not this way it does not work unless I do it as I did in the previous video.

Be that as it may, it is a bit annoying because I already know that it is logical that the app has to be started again so that it uses the `initState()` method and thus the StatefulWidget is loaded correctly, but this did not do it before. If I simply threw the error in the console without stopping everything and then executing a hot restart would restart the application and everything would continue working, it would be a great advance.

**Comment by darshankawar** [comment]
@hekhy 
Thanks for these details. I tried to follow all the steps you mentioned as below:

1. Ran flutter counter app.
2. Changed `StatelessWidget` to `Stateful` and saved the changed.
3. Had breakpoint at `MyApp` class.
4. Hot restart which gave me errors, as `MyApp` class didn't satisfy the `stateful` class rules as below:

![test1](https://user-images.githubusercontent.com/67046386/135218116-e846c39f-06b4-4219-9eac-9ce66eef27d9.png)

Are you purely trying these in debugging mode ?

**Comment by hekhy** [comment]
So, I always start the app in debug mode

**Comment by darshankawar** [comment]
Thanks. Keeping this issue open for further insights from the team.

**Comment by darshankawar** [comment]
Seems related / similar to https://github.com/flutter/flutter/issues/89788

**Comment by hekhy** [comment]
> Gracias. Mantener este problema abierto para obtener más información del equipo.

There is no reason, I want him to fix himself as much as you do.
